### PR TITLE
Move to using the new @pulumi/docker package.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
The Docker build and push functionality has been moved into the `@pulumi/docker` package, so that it can be reused in other contexts as well - such as  Kubernetes, Azure, OpenFaas, etc.  

This change migrates `@pulumi/cloud-aws` to use the copy of this functionality in the new `@pulumi/docker` library.

### CODE REVIEW

File: aws/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
         "typescript": "^2.6.2"
     },
     "peerDependencies": {
-        "@pulumi/cloud": "${VERSION}"
+        "@pulumi/cloud": "${VERSION}",
+        "@pulumi/docker": "${VERSION}"
```
Comment (CyrusNajmabadi) [inline aws/package.json]: this needs to be a real dependency.  but we need @pulumi/docker actually publishing first.  on @ellismg to get that up and running today.

File: aws/package.json [hunk 2]
```diff
@@ -13,6 +13,8 @@
     "dependencies": {
         "@pulumi/aws": "^0.14.0",
         "@pulumi/aws-infra": "^0.14.0",
+        "@pulumi/aws-serverless": "^0.14.0",
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply aws/package.json]: Sorry, part of another change.  will undo.

File: aws/service.ts [hunk 3]
```diff
@@ -21,8 +21,8 @@ import * as assert from "assert";
 import * as stream from "stream";
 import { CloudCluster, CloudNetwork } from "./shared";
 
+import * as docker from "@pulumi/docker";
 import * as config from "./config";
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply aws/service.ts]: Sure.

**Comment by lukehoban** [comment]
@ellismg @CyrusNajmabadi Do we have a plan to publish the `@pulumi/docker` package and merge these changes?

**Comment by ellismg** [comment]
> @ellismg @CyrusNajmabadi Do we have a plan to publish the @pulumi/docker package and merge these changes?

It is now published and public https://www.npmjs.com/package/@pulumi/docker

**Comment by ellismg** [comment]
Pushed ad5a46e15b479776884b7db79fb490e6abe6a4b1 which updates the lockfile and removed the unnecessary aws-serverless reference.

**Pull request merged**